src/otx/core/metrics/pck.py

```diff
@@ -190,19 +178,33 @@ def update(self, preds: list[dict[str, Tensor]], target: list[dict[str, Tensor]]
     def compute(self) -> dict:
         """Compute PCK score metric."""
         pred_kpts = np.stack([p[0].cpu().numpy() for p in self.preds])
-        gt_kpts = np.stack([p[0] for p in self.targets])
-        kpts_visible = np.stack([p[1] for p in self.targets])
-
-        normalize = np.tile(np.array([self.input_size]), (pred_kpts.shape[0], 1))
+        gt_kpts_processed = []
+        for p in self.targets:
+            if len(p[0].shape) == 3 and p[0].shape[0] == 1:
+                gt_kpts_processed.append(p[0].squeeze())
+            else:
+                gt_kpts_processed.append(p[0])
+        gt_kpts = np.stack(gt_kpts_processed)
+
+        kpts_visible = []
+        for p in self.targets:
+            if len(p[1].shape) == 3 and p[1].shape[0] == 1:
+                kpts_visible.append(p[1].squeeze())
+            else:
+                kpts_visible.append(p[1])
+
+        kpts_visible_stacked = np.stack(kpts_visible)
+
+        normalize = np.tile(np.array([self.input_size[::-1]]), (pred_kpts.shape[0], 1))
```

Comment:
the reason why input size order is reversed is that norm_factor order is (w, h)? Referring docstring of the function, it should be (h, w). Which one is right?

Reply:
Right, it's wh, since we use xy keypoints coordinates. The function is order-agnostic, we can use it for both xy and xy. The main point is that we need to pass aligned kps and normalization, which are in xy format for our case.

Reply:
Thanks for explanation. Could you update `norm_factor` docstring at `keypoint_pck_accuracy` then?